src/plugins/query_enhancements/public/datasets/s3_type.ts

```diff
@@ -246,3 +263,63 @@ const fetchTables = async (http: HttpSetup, path: DataStructure[]): Promise<Data
 
   return fetch(http, path, 'TABLE');
 };
+
+// Function to process the input and map types using the new SQL to OSD mapping
+export function mapResponseToFields(sqlOutput: SQLQueryResponse): DatasetField[] {
+  const datasetFields: DatasetField[] = [];
+
+  for (const row of sqlOutput.datarows) {
+    const [colName, dataType] = row;
+
+    // Stop processing once we hit the partition info row
+    if (colName === S3_PARTITION_INFO_COLUMN) {
+      break;
+    }
+
+    // Only include rows with valid types
+    if (dataType && dataType !== '') {
```

Comment:
```suggestion
    if (dataType) {
```
Is `dataType` always a string?

Reply:
if `(dataType && dataType !== '')` specifically excludes empty strings.